From ad7c02daca5b3a97f274b78fada42af27b156635 Mon Sep 17 00:00:00 2001
From: Stephan Holljes <klaxa1337@googlemail.com>
Date: Mon, 10 Apr 2017 06:50:15 +0200
Subject: [PATCH 3/3] lavf/tcp.c: Free allocated client URLContext in case of
 error.

---
 libavformat/tcp.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libavformat/tcp.c b/libavformat/tcp.c
index 07b4ed9..f1a597d 100644
--- a/libavformat/tcp.c
+++ b/libavformat/tcp.c
@@ -203,8 +203,10 @@ static int tcp_accept(URLContext *s, URLContext **c)
         return ret;
     cc = (*c)->priv_data;
     ret = ff_accept(sc->fd, sc->listen_timeout, s);
-    if (ret < 0)
+    if (ret < 0) {
+        ffurl_close(*c);
         return ret;
+    }
     cc->fd = ret;
     return 0;
 }
-- 
2.9.3

